### PR TITLE
fix(sdk-middleware-http): try json parse with catch block

### PR DESCRIPTION
**Pull request by markus-azer**
#### Summary

wrap json.parse with try and catch

#### Description

Wrapping it so in case we receive any invalid json, instead of directly parsing it which could lead to `Unexpected end of JSON input`, in this case, if there is any error then it get catch and then forward `body` to consumer.

### CODE REVIEW

**Review by daern91 — APPROVED** [review]
LGTM 👍

**Review by markus-azer — COMMENTED** [review]
@daern91 any idea why it works with 11 ?

File: integration-tests/cli/helpers/personal-data-erasure.data.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ export const customLineItem = [
         slug: 'mySlug',
         taxCategory: {
           typeId: 'tax-category',
-          id: 'fd890829-7554-4c22-a806-d6c6dcdbb0c5',
+          id: 'd205cc42-e399-424a-b6fc-0ef44772d6bc',
```
Comment (markus-azer) [inline integration-tests/cli/helpers/personal-data-erasure.data.js:66]: changed it cause i accidentally deleted it

File: integration-tests/cli/personal-data-erasure.it.js [hunk 2]
```diff
@@ -105,7 +105,19 @@ describe('personal data erasure', () => {
...
+          expect(data).toContainEqual(expect.objectContaining({type: 'CustomerCreated'}))
+          expect(data).toContainEqual(expect.objectContaining({type: 'ReviewCreated'}))
+          expect(data).toContainEqual(expect.objectContaining({type: 'OrderCreated'}))
+          expect(data).toContainEqual(expect.objectContaining({type: 'CartCreated'}))
+          expect(data).toContainEqual(expect.objectContaining({type: 'Order'}))
+          expect(data).toContainEqual(expect.objectContaining({type: 'Cart'}))
+          expect(data).toContainEqual(expect.objectContaining({email: 'foo@bar.de'}))
+          expect(data).toContainEqual(expect.objectContaining({amountPlanned: { type: 'centPrecision', currencyCode: 'EUR', centAmount: 100, fractionDigits: 2}}))
+          expect(data).toContainEqual(expect.objectContaining({name: {de: 'deutscherListenName', en: 'englishListName'}}))
+          expect(data).toContainEqual(expect.objectContaining({text: 'Review text'}))
```
Comment (markus-azer) [inline integration-tests/cli/personal-data-erasure.it.js:120]: these checks to further be easy if new object were introduced

**Review by daern91 — APPROVED** [review]

**Pull request merged**